### ISSUE TITLE
Incorrect "in_app" processing of sourcemaps that have a webpack "namepsace"

### ISSUE THREAD

**Issue by ob1** [getsentry/sentry#20745]
## Description

My app's stack frames are marked as non-`in_app` due to the webpack-specific sourcemap processing at https://github.com/getsentry/sentry/blob/d54f3cac09cf6925204a52d43b7ddec510ec3a6b/src/sentry/lang/javascript/processor.py#L695

This code assumes a `webpack:///` prefix, while code that is packaged as a library (or otherwise sets a "namespace") will have a `webpack://[namespace]/` prefix.

See https://webpack.js.org/configuration/output/#outputdevtoolmodulefilenametemplate

### What you expected to happen

Frames should remain marked `in_app`: true

### Possible Solution

Take into account the general case format: `webpack://[optional namespace]/`

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-1817) by [Unito](https://www.unito.io)

**Comment by ob1** [comment]
Please consider this issue.
It has even been offered a solution via https://github.com/getsentry/sentry/pull/21718

**Comment by Kranie** [comment]
Hi fellow Sentrycians,

Im sorry for text of doom, I am sitting in a blind spot and this could possibly be the reason and also worth a bump.

I've been struggling to implement Sentry in our new front end environment.
We are currently converting most of necessary repositories into packages sitting in a monorepo using Yarn v2.

That means our hosted websites now sits `<namespace>/<package>` in a single repository.
We did have sentry on-prem before working with sourcemaps correctly demystifying non-sense minified code to its source, and are currently working on moving to hosted sentry for the new monorepo setup.

Before the transition to monorepo we used the sentry-cli to upload the source-maps for the given website, we decided to move along and use the sentry webpack-plugin. 

At first I tried the minimal approach based on the example from the docs when using react. Artifacts where uploaded just fine and no "red"-box in the Sentry ui. However the error stacktracing showed up as the minified code.
 
From there I basicly tried to mimic the behaviour of the earlier setup, as i could see that sentry webpack-plugin basicly just invokes sentry-cli with arguments referencing the options given to the sentry webpack-plugin.

After notifying this release and uploading new artifacts I see a "red"-box in the Sentry ui, telling me:

_There were 2 errors encountered while processing this event_
- Source code was not found
**Url** webpack://@skytdc/ssfteams/bundle/app.959c32be91c409b47578.js
- Source code was not found
**Url** webpack://@skytdc/ssfteams/bundle/21.ee89d5a7f256279ed930.js 

Given the topic of this issue it seems to reflect my problem. It is worth mentioning that how source maps references and how Sentry works internally is not really a subject I know a lot about :upside_down_face:

As ending words, I have been chiming docs up and down and googling for this. Any push in the right direction would be absolutely awesome. And if this reflects the issue commented upon even better!
I have added some TLDR; section that might be irrelevant for the issue idk.
TLDR;
While transitioning we also also implemented typescript and we use bable-loader, webpack was upgraded from 4 to 5 and the devtools option in webpack production configuration was changed from "source-maps" to "cheap-module-source-map", this was due to docker build issue causing "javascript heap out of memory" when using "source-maps".

Upon debugging vartions of the options uploading artifacts to sentry, I stumbled on "red"-box error in the Sentry ui, while attempting to setup locally. Where Sentry would attempt to fetch the given resource file on localhost? The concerning file it tried to fetch was already upload on the realease as artifact, this was pretty clear as the files we produce of webpack output are suffixed with the content hash of the file `filename: 'bundle/[name].[contenthash].js'` -- I can't seems to find any pointer as of to why this fetch happens. but it does seems that it means that some part of the setup is correct when it attempts to do so. I really don't remember the setting up of sentry to be all this fuzzy

**Comment by Kranie** [comment]
Just leave an update on my above comment.

we flipped the devtools option back to "source-maps" and fixed the issue of memory. in terms of source maps we decided to go for "scrapping" (source-maps open public) our  instead of "private" and uploading the artifacts to sentry server. this fixed the problems we had when attempting to integrate with sentry (at least without pushing artifacts/sourcemaps to sentry). 

Im sorry if I was hijacking the issue! And if my comment was misplaced. Still I dont know if we ran into some of the same issues with the namespacing and private source-maps, issue seems very relevant still.

**Comment by kamilogorek** [comment]
@Kranie thanks for the feedback and getting back with the current update. I believe this could be fixed with a mix of `strip-prefix`, `url-prefix` options and `RewriteFrames` sdk integration.

@ob1 reviewed your PR, thanks for the contribution

**Comment by kamilogorek** [comment]
We're moving all processing to symbolicator now, and this should be fixed there:
https://github.com/getsentry/symbolicator/blob/master/crates/symbolicator-service/src/services/symbolication/js.rs#L392C15-L393

Once we make it GA, we can get back to this issue and see if it persists.

**Comment by roggenkemper** [comment]
Closing as this is expected to be fixed - if the issue persists, feel free to re-open
